client/post-editor/post-editor.jsx

```diff
@@ -1013,17 +1014,268 @@ export const PostEditor = React.createClass( {
 		return editorMode;
 	},
 
+	isCursorInATag: function( content, cursorPosition ) {
+		const lastLtPos = content.lastIndexOf( '<', cursorPosition );
+		const lastGtPos = content.lastIndexOf( '>', cursorPosition );
+
+		if ( lastLtPos > lastGtPos ) {
+			// inside a tag that was opened, but not closed
+
+			// find what the tag is
+			const tagContent = content.substr( lastLtPos );
+			const tagMatch = tagContent.match( /<\s*(\/)?(\w+)/ );
+			if ( ! tagMatch ) {
+				return null;
+			}
+
+			const tagType = tagMatch[ 2 ];
+			const closingGt = tagContent.indexOf( '>' );
+			const isClosingTag = ! ! tagMatch[ 1 ];
```

Comment:
Good catch! :) PHPStorm is a bit flaky with some odd syntax like this. It's very possible the other instances of that are because of the same reason :)